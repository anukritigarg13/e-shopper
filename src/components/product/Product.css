.product{
    border: solid thin gray;
    border-radius: 10px;
    margin: 1%;
    padding: 10px;
    text-align: center;
    width: 20%;
    box-shadow:10px 10px 5px grey;
}
.quantity-placard{
    background-color: rgb(221, 213, 213);
    padding: 10px;
}
.product-image{
    width: 100%
}
.product-image-div{
    height: 7rem;
    display: flex;
    justify-content: center;
}

@media screen and (max-width:427px){
    .product{
        width: 80%;
    }
}