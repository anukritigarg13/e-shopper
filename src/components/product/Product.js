import React from 'react';
import PropTypes from 'prop-types';
import QuantityControl from '../QuantityControl/QuantityControl';
import './Product.css';

const Product = (props) => (
  <div className="product">
    <div className="product-image-div">
      <img className="product-image" src={props.imgSrc} alt={props.imgAlt} />
    </div>

    <p>{props.companyName}</p>
    <p>{props.itemName}</p>
    <p>{props.unitQuantity}</p>
    <div className="quantity-placard">
      <p>
        MRP
        {' '}
        {props.unitPrice}
        /-
      </p>
      <QuantityControl
        add={props.add}
        remove={props.remove}
        quantity={props.itemCount}
      />
    </div>
  </div>
);

Product.propTypes = {
  imgAlt: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  itemName: PropTypes.string.isRequired,
  unitPrice: PropTypes.number.isRequired,
  unitQuantity: PropTypes.string.isRequired,
  add: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
  itemCount: PropTypes.number.isRequired,

};
export default Product;
