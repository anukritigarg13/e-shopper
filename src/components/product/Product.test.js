/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import Product from './Product';

describe('Product component', () => {
  let mockProps;
  beforeAll(() => {
    mockProps = {
      imgSrc: 'MOCK_IMG_SRC',
      imgAlt: 'MOCK_IMG_ALT',
      companyName: 'MOCK_COMPANY_NAME',
      itemName: 'MOCK_ITEM_NAME',
      itemCount: 3,
      unitQuantity: '1 Kg',
      unitPrice: 40,
      stock: 50,
      add: jest.fn(),
      remove: jest.fn(),
    };
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should display product image', () => {
    render(<Product {...mockProps} />);
    const imgElement = screen.getByTestId('product-img');
    expect(imgElement.tagName).toBe('IMG');
  });
  it('should display the QuantityControl component', () => {
    render(<Product {...mockProps} />);
    screen.getByTestId('quantity-control');
  });
  it('should display company name, itemName and unitQuantity', () => {
    render(<Product {...mockProps} />);
    screen.getByText('MOCK_COMPANY_NAME');
    screen.getByText('MOCK_ITEM_NAME');
    screen.getByText('1 Kg');
  });
  it('should display unit price', () => {
    render(<Product {...mockProps} />);
    screen.getByText('MRP 40/-');
  });
  it('should match snapshot', () => {
    const { container } = render(<Product {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
