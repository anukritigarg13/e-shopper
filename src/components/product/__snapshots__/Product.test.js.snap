// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should match snapshot 1`] = `
<div>
  <div
    class="product"
    data-testid="product-test"
  >
    <div
      class="product-image-div"
    >
      <img
        alt="MOCK_IMG_ALT"
        class="product-image"
        data-testid="product-img"
        src="MOCK_IMG_SRC"
      />
    </div>
    <p>
      MOCK_COMPANY_NAME
    </p>
    <p>
      MOCK_ITEM_NAME
    </p>
    <p>
      1 Kg
    </p>
    <div
      class="quantity-placard"
    >
      <p>
        MRP
         
        40
        /-
      </p>
      <div
        class="quantity-control"
        data-testid="quantity-control"
      >
        <p>
          <button
            class="control-button"
            type="button"
          >
            +
          </button>
        </p>
        <p
          class="quantity-in-basket"
        >
          3
           
          in Basket
        </p>
        <p>
          <button
            class="control-button"
            type="button"
          >
            -
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
